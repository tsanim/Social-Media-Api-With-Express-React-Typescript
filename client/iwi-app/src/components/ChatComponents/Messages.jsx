import React from 'react';
import ScrollToBottom from 'react-scroll-to-bottom';
import PropTypes from 'prop-types';
import Message from './Message';

function Messages(props) {
    return (
        <ScrollToBottom className="messagesContainer">
            <p className="msg">{props.infoMessage}</p>
            <ul>
                {
                    props.messages.map((m, i) => {
                        let liClass = i === 0 ? 'clearfix' : null;
                        let isMine = m.creator._id === localStorage.getItem('userId');

                        return (
                            (<li key={i} className={liClass}>
                                <Message 
                                    isMine={isMine}
                                    message={m}
                                />
                            </li>)
                        )
                    })
                }
            </ul>
        </ScrollToBottom>
    )
}

Messages.propTypes = {
    infoMessage: PropTypes.string,
    messages: PropTypes.array,
}

export default Messages;