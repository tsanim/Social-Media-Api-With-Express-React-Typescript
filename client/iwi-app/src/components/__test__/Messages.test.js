import React from 'react';
import Messages from '../ChatComponents/Messages';
import { shallow } from 'enzyme';
import ScrollToBottom from 'react-scroll-to-bottom';

function messagesComponentSetup() {
    const props = {
        infoMessage: '',
        messages: [ { creator: {} } ],
    }

    const enzymeWrapper = shallow(<Messages {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('Messages component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = messagesComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render ScrollToBottom Component with class "messagesContainer"', () => {
        const { enzymeWrapper } = messagesComponentSetup();

        expect(enzymeWrapper.find(ScrollToBottom).hasClass('messagesContainer')).toEqual(true);
    });

    it('should render p tag with class "msg"', () => {
        const { enzymeWrapper } = messagesComponentSetup();

        expect(enzymeWrapper.find('p').hasClass('msg')).toEqual(true);
    });

    it('should render ul tag with items count equal to messages length', () => {
        const { enzymeWrapper, props } = messagesComponentSetup();

        expect(enzymeWrapper.find('ul').children()).toHaveLength(props.messages.length);
    });

    it('should render ul tag with first item with class "clearfix"', () => {
        const { enzymeWrapper } = messagesComponentSetup();

        expect(enzymeWrapper.find('ul').childAt(0).hasClass('clearfix')).toEqual(true);
    });
});