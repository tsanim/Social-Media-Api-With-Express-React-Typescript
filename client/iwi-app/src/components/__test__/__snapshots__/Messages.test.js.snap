// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component should render initial layout 1`] = `
Array [
  <_default
    className="messagesContainer"
  >
    <p
      className="msg"
    >
      
    </p>
    <ul>
      <li
        className="clearfix"
      >
        <Message
          isMine={false}
          message={
            Object {
              "creator": Object {},
            }
          }
        />
      </li>
    </ul>
  </_default>,
]
`;
